feat: Add slug to collections and use it in public collection URLs

tw4l opened the pull request:
Resolves https://github.com/webrecorder/browsertrix/issues/2298

## Changes

- Slugs added to collections, can be specified separately when creating or updating collections or else is based off of supplied collection name
- Migration added to backfill slugs for existing collections
- Redirect collection to newest slug if changed
- Adds option to copy public profile link to "Public Collections" action menu
- Show "Back to <Org>" link instead of breadcrumbs

## Manual testing

1. Log in as crawler
2. Create or go to a public collection
3. Click "Copy Link"
4. Paste link into new tab. Verify collection loads as expected
5. Click "< Back to <org name>". Verify org profile loads as expected
6. Click collection from org profile. Verify collection loads as expected
7. Click pencil icon to edit collection
8. Update and save new collection name
9. Repeat 3-4. Verify collection loads with new slug
10. Go back to previous tab and copy link with old slug
11. Open new incognito window and paste link with old slug. Verify you're redirected to the newest slug
12. Go to logged in tab and create a new public collection using the old slug. Verify collection can be created and public view loads as expected

## Screenshots

| Page | Image/video |
| ---- | ----------- |
| Dashboard | <img width="232" alt="Screenshot 2025-01-13 at 8 33 25 PM" src="https://github.com/user-attachments/assets/a88144d5-46cf-40f0-80c5-c5678d51931b" /> |
| Public collection | <img width="507" alt="Screenshot 2025-01-13 at 8 33 43 PM" src="https://github.com/user-attachments/assets/b81797b2-a349-405e-824b-5baf6747a9f0" /> |

tw4l committed:
Add slugs to collections and backfill via migration

tw4l committed:
Use slugs for public coll urls, support redirects from old slugs

In case of redirect, add redirectToSlug field to response with new
slug to make it simple for frontend to pick up on redirect.

tw4l committed:
Fix broken tests

tw4l committed:
Add slug to collections on org import if not already present

tw4l committed:
Fix regression with duplicate org sub response

SuaYoo committed:
rename org slug

SuaYoo committed:
update route

SuaYoo committed:
update breadcrumbs

SuaYoo committed:
fix replay

SuaYoo committed:
add redirect

SuaYoo committed:
update org slug

SuaYoo committed:
track collection by slug

SuaYoo committed:
use redirect value

SuaYoo committed:
add link to copy share

SuaYoo commented:
@tw4l tested backend locally, looks good--frontend ready for review

tw4l committed:
Rename migration to 0039

ikreymer commented on backend/btrixcloud/colls.py:
```diff
@@ -1044,12 +1127,14 @@ async def download_public_collection(
             raise HTTPException(status_code=403, detail="not_allowed")
 
-        return await colls.download_collection(coll_id, org)
+        return await colls.download_collection(cast(UUID, coll.id), org)
```
Is this because it could be null also?
I think we've added a `id: UUID` to some models to avoid needing to cast, like in User. Should we add it to collection to?

tw4l replied:
Yeah it's because the BaseMongoModel makes it so that `id` can be UUID, str, or None. Adding `id: UUID` to the collection models would be a better way to handle this I think! In practice they're already being created as UUIDs. Will make that change.

tw4l replied:
Made this change and rebased on main :)

tw4l committed:
Change id typing for collections to always be UUID

tw4l committed:
Remove new slug from previousSlugs of other collections in same org

ikreymer commented:
~~The collection loads under the old slug, but the URL is not redirect, still has the old slug. Verified the API returns the new slug under `slug`.~~
Nevermind, didn't have latest!

ikreymer commented on frontend/src/pages/collections/collection.ts:
```diff
@@ -48,27 +48,36 @@ export class Collection extends BtrixElement {
+        collectionSlug,
+      });
+
+      if (collection.redirectToSlug) {
```
I think we don't actually need redirectToSlug at all, since can simply check this instead:

```suggestion
  if (collection.slug !== collectionSlug) {
```

Will simplify!

ikreymer replied:
Updated in latest commit!

ikreymer committed:
simplify: remove redirectToSlug from backend and frontend, frontend can determine if redirect is needed
by comparing the slug in the API request to the 'slug' in the response, and redirect if they're different
